extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Subscriptions Block.
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Subscriptions;
+
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'subscriptions';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block( BLOCK_NAME );
```

Comment:
I'd recommend checking if the Subscriptions module is active before you register the block, since it can't really be used without the module. On WordPress.com, however, it can be used all the time, so you could check for one or the other.

```suggestion
	if (
		( Jetpack::is_active() && Jetpack::is_module_active( 'subscriptions' ) )
		|| ( defined( 'IS_WPCOM' ) && IS_WPCOM )
	) {
		jetpack_register_block( BLOCK_NAME );
	}
```